### ISSUE TITLE
TypeError: descriptor '__subclasses__' of 'type' object needs an argument when imported in python 3.5.2

### ISSUE THREAD

**Issue by BenBBear** [tmbo/questionary#8]
This is a known bug in Python 3.5.2 as discussed in https://stackoverflow.com/questions/42942867/optionaltypefoo-raises-typeerror-in-python-3-5-2

```
~/.local/lib/python3.5/site-packages/questionary/prompts/common.py in <module>
    246 def build_validator(validate: Union[Type[Validator],
    247                                     Callable[[Text], bool],
--> 248                                     None]
    249                     ) -> Optional[Validator]:
    250     if validate:

/usr/lib/python3.5/typing.py in __getitem__(self, parameters)
    550             parameters = (parameters,)
    551         return self.__class__(self.__name__, self.__bases__,
--> 552                               dict(self.__dict__), parameters, _root=True)
    553 
    554     def __eq__(self, other):

/usr/lib/python3.5/typing.py in __new__(cls, name, bases, namespace, parameters, _root)
    510                 continue
    511             if any(isinstance(t2, type) and issubclass(t1, t2)
--> 512                    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
    513                 all_params.remove(t1)
    514         # It's not a union if there's only one type left.

/usr/lib/python3.5/typing.py in <genexpr>(.0)
    510                 continue
    511             if any(isinstance(t2, type) and issubclass(t1, t2)
--> 512                    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
    513                 all_params.remove(t1)
    514         # It's not a union if there's only one type left.

/usr/lib/python3.5/typing.py in __subclasscheck__(self, cls)
   1075                     return True
   1076                 # If we break out of the loop, the superclass gets a chance.
-> 1077         if super().__subclasscheck__(cls):
   1078             return True
   1079         if self.__extra__ is None or isinstance(cls, GenericMeta):

/usr/lib/python3.5/abc.py in __subclasscheck__(cls, subclass)
    223                 return True
    224         # Check if it's a subclass of a subclass (recursive)
--> 225         for scls in cls.__subclasses__():
    226             if issubclass(subclass, scls):
    227                 cls._abc_cache.add(subclass)

TypeError: descriptor '__subclasses__' of 'type' object needs an argument

```

Do you consider comment that type hint `->Optional[Validator]` out for Python 3.5.2 compatibility so more people can use this wonderful library?  Thanks a lot.

**Comment by tmbo** [comment]
Yes sure will do :+1:

**Comment by tmbo** [comment]
should be fixed in version `1.1.1`
